File: PublicServiceRequest/Models/ServiceTypeOwnerGroup.cs
namespace CustomerService_PSR.Models
{
    using System;
    using System.Collections.Generic;
    using System.ComponentModel.DataAnnotations;
    using System.ComponentModel.DataAnnotations.Schema;
    using System.Data.Entity.Spatial;

    [Table("PSR.ServiceTypeOwnerGroup")]
    public partial class ServiceTypeOwnerGroup : BaseClass
    {
        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Usage", "CA2214:DoNotCallOverridableMethodsInConstructors")]
        public ServiceTypeOwnerGroup()
        {
            ServiceTypes = new HashSet<ServiceType>();
            ServiceTypesForOverridingOwnerGroup = new HashSet<ServiceType>();
            Requests = new HashSet<Request>();
            RequestActions = new HashSet<RequestAction>();
        }
        public Guid? ServiceTypeOwnerGroupID { get; set; }

        public int? ServiceTypeOwnerGroupLocationID { get; set; }

        [Required]
        [StringLength(100)]
        public string ServiceTypeOwnerGroupName { get; set; }

        [StringLength(500)]
        public string ServiceTypeOwnerGroupDescription { get; set; }

        [StringLength(1000)]
        public string ServiceTypeOwnerAvailableToActiveDirectoryGroupName { get; set; }

        [StringLength(1000)]
        public string ServiceTypeOwnerActiveDirectoryGroupName { get; set; }

        [StringLength(500)]
        public string ServiceTypeOwnerGroupMainEmail { get; set; }

        [StringLength(500)]
        public string ServiceTypeOwnerGroupNotificationEmail { get; set; }

        [StringLength(500)]
        public string ServiceTypeOwnerGroupEscalationEmail { get; set; }

        [StringLength(25)]
        public string ServiceTypeOwnerGroupPhoneNo { get; set; }

        public int? ServiceTypeOwnerGroupDistrictNo { get; set; }

        public bool? SelectableOnRequestCreationFlag { get; set; }

        public bool? SelectableOnRequestReassignmentFlag { get; set; }

        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Usage", "CA2227:CollectionPropertiesShouldBeReadOnly")]
        public virtual ICollection<Request> Requests { get; set; }

        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Usage", "CA2227:CollectionPropertiesShouldBeReadOnly")]
        public virtual ICollection<RequestAction> RequestActions { get; set; }

        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Usage", "CA2227:CollectionPropertiesShouldBeReadOnly")]
        public virtual ICollection<ServiceType> ServiceTypes { get; set; }

        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Usage", "CA2227:CollectionPropertiesShouldBeReadOnly")]
        public virtual ICollection<ServiceType> ServiceTypesForOverridingOwnerGroup { get; set; }

        public virtual ServiceTypeOwnerGroupLocation ServiceTypeOwnerGroupLocation { get; set; }
    }
}

File: PublicServiceRequest/app/domain/serviceTypeOwnerGroup.ts
module app.domain {

    export interface IServiceTypeOwnerGroup {
        ServiceTypeOwnerGroupID: number;
        ServiceTypeOwnerGroupLocationID: number;
        ServiceTypeOwnerGroupName: string;
        ServiceTypeOwnerGroupDescription: string;
        ServiceTypeOwnerGroupPhoneNo: string;
        ServiceTypeOwnerGroupDistrictNo: number;
        ServiceTypeOwnerGroupNotificationEmail: string;
        ServiceTypeOwnerGroupEscalationEmail: string;
        ServiceTypeOwnerGroupMainEmail: string;
        AdmIsActive: string;
        AdmUserAdded: number;
        AdmUserAddedFullName: string;
        AdmDateAdded: string;
        AdmUserModified: number;
        AdmUserModifiedFullName: string;
        AdmDateModified: string;
    }

    export class ServiceTypeOwnerGroup implements IServiceTypeOwnerGroup {
        constructor(public ServiceTypeOwnerGroupID: number,
            public ServiceTypeOwnerGroupLocationID: number,
            public ServiceTypeOwnerGroupName: string,
            public ServiceTypeOwnerGroupDescription: string,
            public ServiceTypeOwnerGroupPhoneNo: string,
            public ServiceTypeOwnerGroupDistrictNo: number,
            public ServiceTypeOwnerGroupNotificationEmail: string,
            public ServiceTypeOwnerGroupEscalationEmail: string,
            public ServiceTypeOwnerGroupMainEmail: string,
            public AdmIsActive: string,
            public AdmUserAdded: number,
            public AdmUserAddedFullName: string,
            public AdmDateAdded: string,
            public AdmUserModified: number,
            public AdmUserModifiedFullName: string,
            public AdmDateModified: string) { }
    }
}